### ISSUE TITLE
What is the format of image matching label

### ISSUE THREAD

**Issue by InstantWindy** [ubc-vision/image-matching-benchmark#23]
Hi. Is there any label formatting instructions for the dataset. I'm a little confused about what is recorded in the depth_map folder, calibration folder, new-vis-pairs folder, and visiability folder under the set_100 folder in the dataset. Thanks.

**Comment by etrulls** [comment]
Those files are for validation, so you don't need to interact with them. You should just place them in the right folder and the benchmark will run; you can check the benchmark code to see how they are used, if you're interested.

There are examples about how to process depth maps, calibration files, and co-visibility estimates for the **training** data in [this notebook](https://github.com/vcg-uvic/image-matching-benchmark/blob/master/example/training_data/parse_data.ipynb). The format for the validation files might be a bit different.

Does that answer your question?

**Comment by InstantWindy** [comment]
Thank you very much. But I don't understand why to load pairs from the new-vis-pairs/keys-th-{:0.1f}.npy file in the compute_match.py file. What does this new-vis-pairs/keys-th-{:0.1f}.npy file mean?

**Comment by etrulls** [comment]
That file simply contains co-visibility estimates for every possible combination of two images.

You can download the files [here](https://vision.uvic.ca/imw-challenge/index.md) ("validation"/"all sequences") and inspect them yourself.

You can check how they are being used [here](https://github.com/vcg-uvic/image-matching-benchmark/blob/3e9709e0f21351cebd2e20d7700222554d59eeba/utils/path_helper.py#L537) (check where that function is being called and what it is used for).

But as I said earlier, you don't need to interact with this file at all. All stereo metrics are compiled at different co-visibility thresholds for you, if you care to look at them. You can see this in the example result file [here](https://github.com/vcg-uvic/image-matching-benchmark/blob/fb846a6cb53ea034209fea3b36038d943bd767d7/example/results/val/rootsift-upright-2k-degensac-inlier-th-0.5.json#L902). See all the results with `_th_<covisibility_value>`. The paper ([here](https://drive.google.com/open?id=13kmXJKXWS_xci0L-9tnbgwy-Y2AXVGco); link to gdrive because we've updated the arxiv paper but it's not up yet) contains examples: see sections 3 and 6.

I can't help you beyond that if I don't understand what you're trying to do.

**Comment by InstantWindy** [comment]
Thanks.I just want to know the label format of the image matching used in the training, because I want to train from scratch with my own dataset.

**Comment by etrulls** [comment]
Then you need to use the labels from the training set which are in a different format, with examples in the notebook I sent in my first reply (cells 30+).
